doc/reference/reference_lua/datetime.rst

```diff
@@ -1,62 +1,958 @@
 It allows creating the date and time values either via the object interface
 or via parsing string values conforming to the ISO-8601 standard.
 
-Below is a list of the ``datetime`` module functions and methods.
+.. _interval_arithm:
 
-.. container:: table
+Interval arithmetic
+-------------------
+
+The ``datetime`` module enables creating of objects of two types: ``datetime`` and ``interval``.
+
+If you need to shift the ``datetime`` object values, you can use either the modifier methods, that is, the :ref:`datetime_object:add() <datetime-add>` or :ref:`datetime_object:sub() <datetime-sub>` methods,
+or apply interval arithmetic using overloaded `+ (__add)` or `- (__sub)` methods.
+
+``datetime_object:add()``/``datetime_object:sub()`` modify the current object, but ``+``/``-`` create copy of the object as the operation result.
+
+In the interval operation, each of the interval subcomponents is sequentially calculated starting from the largest (``year``) to the smallest (``nsec``):
+
+*   ``year`` -- years
+*   ``month`` -- months
+*   ``week`` -- weeks
+*   ``day`` -- days
+*   ``hour`` -- hours
+*   ``min`` -- minutes
+*   ``sec`` -- seconds
+*   ``nsec`` -- nanoseconds
+
+If the results of the operation exceed the allowed range for any of the components, an exception is raised.
+
+The ``datetime`` and ``interval`` objects can participate in arithmetic operations:
+
+*   The sum of two intervals is an interval object, whose fields are the sum of each particular component of operands.
+
+*   The result of subtraction of two intervals is similar: it's an interval object where each subcomponent is the result of subtraction of particular fields in the original operands.
+
+*   If you add datetime and interval objects, the result is a datetime object. The addition is performed in a determined order from the largest component (``year``) to the smallest (``nsec``).
+
+*   Subtraction of two datetime objects produces an interval object. The difference of two time values is performed not as the difference of the epoch seconds,
+    but as difference of all the subcomponents, that is, years, months, days, hours, minutes, and seconds.
+
+*   An untyped table object can be used in each context where the typed datetime or interval objects are used if the left operand is a typed object with an overloaded operation of ``+`` or ``-``.
+
+The matrix of the ``addition`` operands eligibility and their result types:
+
+..  container:: table
+
+    ..  list-table::
+        :widths: 25 25 25 25
+        :header-rows: 1
+
+        *   -
+            -   datetime
+            -   interval
+            -   table
+
+        *   -   **datetime**
+            -
+            -   datetime
+            -   datetime
+
+        *   -   **interval**
+            -   datetime
+            -   interval
+            -   interval
+
+        *   -   **table**
+            -
+            -
+            -
```

Comment:
If I'm correct, the only valid case here is `table+datetime = datetime`

Reply:
Please see an original matrix here https://github.com/tarantool/tarantool/wiki/Datetime-Internals#arithmetic-operations

> The matrix of the addition operands eligibility and their result types:

As I got it right, the last line can be removed, because combinations with it already exist (see column `table`) and I expect that addition is commutative (`datetime` + `table` == `table` + `datetime` etc). Also, I propose to add `unsupported` for a cell `datetime` + `datetime`.

> The matrix of the subtraction operands eligibility and their result types:

I propose to add `unsupported` for a cell `interval` - `datetime`.

Reply:
Thanks!